### DISCUSSION TITLE
Problems with Calendar Increment/Decrement Buttons

### DISCUSSION THREAD

**Discussion by peteletkemanrakuten** [primefaces/community#3010] [PrimeReact]
![incAndDec](https://github.com/user-attachments/assets/ef16d371-7cf9-4663-b1c7-004c829d9f4d)
With my current setup I can use the Calendar control in PrimeReact to set the date without any issues whatsoever.
However, as soon as I click on the buttons in the increment/decrement buttons the system goes into an indefinite loop.

I've tried what I can in StackBlitz and there is works fine.  
However, this is a company project and I can't put the whole project on StackBlitz or any site like that and it uses internal systems to grab data.

Here is the general flow that seems to work for all other controls, including the date selection, just not for the time selection.

1. In JS I have a JSON array of X input controls, some of them Radio, some of them Selects, some Calendars, etc.
2. On function load JS loops through the array and generates the required controls
    a. loading values from sessionStorage if found
4. The user can then supply input data
5. onChange, a sessionStorage variable is updated, and the correct setState function is called

sessionStorage is used as this is a tabbed interface and most of the time there will be less than six simple values used by the end user.

Are there any pitfalls to using the Calendar control with allowing for time selection and using sessionStorage?

I'm still somewhat new to React and even newer to TypeScript, so I have no doubt that I've made a mistake somewhere.

**Comment by melloware** [comment]
Yeah has to be a bug in your code somewhere if you can't reproduce on StackBlitz and the Showcase is working fine.
